src/once_cell.rs

```diff
@@ -768,42 +760,48 @@ fn now_or_never<T>(f: impl Future<Output = T>) -> T {
     }
     unsafe fn drop(_: *const ()) {}
 
-    futures_lite::pin!(f);
+    // SAFETY: We don't move the future after we pin it here.
+    let future = unsafe { Pin::new_unchecked(&mut f) };
```

Comment:
I strongly recommend that you use the same variable names here. Otherwise, it is very easy to access the owned value incorrectly.

Simple shadowing is not sufficient to prevent misuse, but in this case it is fine because it is on the outermost scope. (This is a common oversight, I sometime see crates [make the false claim that simple shadowing is "doing the exact same thing as pin_mut".](https://github.com/zesterer/pollster/blob/767e6dbabbf93131ad9e7a8935e7a69974e3fd4c/src/lib.rs#L110))

Well, it would be better to duplicate the pin macro as you have done for the ready macro.